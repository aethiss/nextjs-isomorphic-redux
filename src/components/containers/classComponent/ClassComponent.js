import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Styles
import styles from '../../../../styles/Home.module.css'

@connect(
  (state) => ({
    device: state.device,
  }),
  {},
)
class ClassComponent extends Component {
  render() {
    const { device } = this.props
    return (
      <main className={styles.main}>
        <p id={device.isPhone ? 'mobile' : 'desktop'}>
          This is a Server Side Render Page
        </p>
        <p>This example use a React Class Component Connected with Redux</p>
        <p>
          Start edit <code>/components/classComponent/ClassComponent.js</code>{' '}
          to play with this.
        </p>
        <p>
          You are viewing with a {device.isPhone ? 'Mobile' : 'Desktop'}.
          <br />
          {device.isPhone && `Your Mobile is ${device.device}`}
        </p>
        <p>
          please visit{' '}
          <u>
            <a href='https://developers.google.com/speed/pagespeed/insights/?hl=EN'>
              Google Seo Analyze
            </a>
          </u>{' '}
          to test this page
        </p>
      </main>
    )
  }
}

ClassComponent.propTypes = {
  device: PropTypes.object,
}

export default ClassComponent
