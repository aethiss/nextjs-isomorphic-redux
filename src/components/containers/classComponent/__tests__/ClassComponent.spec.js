import React from 'react'
import { Provider } from 'react-redux'
import ClassComponent from '../ClassComponent'

// Tester
import TestRenderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

const desktopProps = {
  device: 'desktop',
  isPhone: false,
}

const mobileProps = {
  device: 'iphone',
  isPhone: true,
}

describe('ClassComponent Component', () => {
  describe('Render', () => {
    it('[ClassComponent] Default render desktop', () => {
      const store = mockStore({
        device: desktopProps,
      })
      const testRenderer = TestRenderer.create(
        <Provider store={store}>
          <ClassComponent />
        </Provider>,
      )
      expect(testRenderer).toMatchSnapshot()
    })
    it('[ClassComponent]: Default render mobile', () => {
      const store = mockStore({
        device: mobileProps,
      })
      const mobileRender = TestRenderer.create(
        <Provider store={store}>
          <ClassComponent />
        </Provider>,
      )
      const instance = mobileRender.root
      expect(mobileRender).toMatchSnapshot()
      expect(instance.findByProps({ id: 'mobile' }).props).not.toBeFalsy()
    })
  })
})
