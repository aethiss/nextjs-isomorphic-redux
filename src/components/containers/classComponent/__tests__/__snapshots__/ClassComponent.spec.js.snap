// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassComponent Component Render [ClassComponent] Default render desktop 1`] = `
<main
  className="main"
>
  <p
    id="desktop"
  >
    This is a Server Side Render Page
  </p>
  <p>
    This example use a React Class Component Connected with Redux
  </p>
  <p>
    Start edit 
    <code>
      /components/classComponent/ClassComponent.js
    </code>
     
    to play with this.
  </p>
  <p>
    You are viewing with a 
    Desktop
    .
    <br />
  </p>
  <p>
    please visit
     
    <u>
      <a
        href="https://developers.google.com/speed/pagespeed/insights/?hl=EN"
      >
        Google Seo Analyze
      </a>
    </u>
     
    to test this page
  </p>
</main>
`;

exports[`ClassComponent Component Render [ClassComponent]: Default render mobile 1`] = `
<main
  className="main"
>
  <p
    id="mobile"
  >
    This is a Server Side Render Page
  </p>
  <p>
    This example use a React Class Component Connected with Redux
  </p>
  <p>
    Start edit 
    <code>
      /components/classComponent/ClassComponent.js
    </code>
     
    to play with this.
  </p>
  <p>
    You are viewing with a 
    Mobile
    .
    <br />
    Your Mobile is iphone
  </p>
  <p>
    please visit
     
    <u>
      <a
        href="https://developers.google.com/speed/pagespeed/insights/?hl=EN"
      >
        Google Seo Analyze
      </a>
    </u>
     
    to test this page
  </p>
</main>
`;
